src/main/java/org/embulk/util/config/Compat.java

```diff
@@ -163,7 +163,7 @@ private static ObjectNode callGetObjectNodeAndRebuildIfAvailable(final DataSourc
         return ObjectNodeRebuilder.rebuild(coreObjectNode, mapper);
     }
 
-    private static Method getToJsonMethod() {
+    private static Method getToJsonMethod(final DataSource source) {
         try {
             // Getting the "toJson" method from embulk-api's public interface "org.embulk.config.DataSource", not from an implementation class,
             // for example "org.embulk.(util.)config.DataSourceImpl", so that invoking the method does not throw IllegalAccessException.
```

Comment:
Does this comment still hold now that we added `dataSourceImplClass.getMethod("toJson");`?

Reply:
Ah, still a kind of valid because we're trying `DataSource#toJson` at first here. Only when it fails, we fallback to get a `Method` instance of an `Impl` class.